spec/ameba/rule/lint/useless_assign_spec.cr

```diff
@@ -1006,6 +1006,47 @@ module Ameba::Rule::Lint
       end
     end
 
+    it "does not report if variable is referenced and there is a deep level scope" do
+      s = Source.new %(
+        response = JSON.build do |json|
+          json.object do
+            json.object do
+              json.object do
+                json.object do
+                  json.object do
+                    json.object do
+                      json.object do
+                        json.object do
+                          json.object do
+                            json.object do
+                              json.object do
+                                json.object do
+                                  json.object do
+                                    json.object do
+                                      anything
+                                    end
+                                  end
+                                end
+                              end
+                            end
+                          end
+                        end
+                      end
+                    end
+                  end
+                end
+              end
+            end
+          end
+        end
+
+        response = JSON.parse(response)
+        response = prepare(response)
```

Comment:
Is this line needed?

```suggestion
```

Reply:
Just to increase a number of variable references and make the recursion cycles to grow.

Reply:
I see, and could it be replaced with some builtin method, like `pointerof` maybe?

Reply:
Ok, i removed it and added one more level of nesting